### PR TITLE
Correct alignment example and documentation

### LINKED ISSUE #11482: Alignment API has changed, documentation is incorrect.
Alignment attributes x2y and y2x have changed to type AlignmentArray and do not have a dataXd member:

```python
from spacy.training import Alignment
​
other_tokens = ["i", "listened", "to", "obama", "'", "s", "podcasts", "."]
spacy_tokens = ["i", "listened", "to", "obama", "'s", "podcasts", "."]
align = Alignment.from_strings(other_tokens, spacy_tokens)
print(f"a -> b, lengths: {align.x2y.lengths}")  # array([1, 1, 1, 1, 1, 1, 1, 1])
print(f"a -> b, mapping: {align.x2y.dataXd}")  # array([0, 1, 2, 3, 4, 4, 5, 6]) : two tokens both refer to "'s"
print(f"b -> a, lengths: {align.y2x.lengths}")  # array([1, 1, 1, 1, 2, 1, 1])   : the token "'s" refers to two tokens
print(f"b -> a, mappings: {align.y2x.dataXd}")  # array([0, 1, 2, 3, 4, 5, 6, 7])

a -> b, lengths: [1 1 1 1 1 1 1 1]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_42/629270707.py in <module>
      5 align = Alignment.from_strings(other_tokens, spacy_tokens)
      6 print(f"a -> b, lengths: {align.x2y.lengths}")  # array([1, 1, 1, 1, 1, 1, 1, 1])
----> 7 print(f"a -> b, mapping: {align.x2y.dataXd}")  # array([0, 1, 2, 3, 4, 4, 5, 6]) : two tokens both refer to "'s"
      8 print(f"b -> a, lengths: {align.y2x.lengths}")  # array([1, 1, 1, 1, 2, 1, 1])   : the token "'s" refers to two tokens
      9 print(f"b -> a, mappings: {align.y2x.dataXd}")  # array([0, 1, 2, 3, 4, 5, 6, 7])

AttributeError: 'spacy.training.alignment_array.AlignmentArray' object has no attribute 'dataXd'
```

Is the `data` attribute on `spacy.training.alignment_array.AlignmentArray` equivalent?

## How to reproduce the behaviour

Run the code in documentation here
https://spacy.io/usage/linguistic-features#aligning-tokenization

## Your Environment

* Operating System: N/A 
* Python Version Used: 3
* spaCy Version Used: 3.4
* Environment Information: via Binder

### PR DESCRIPTION
**Pull request by richardpaulhudson**
## Description

Closes #11482.

### Types of change
Correction to the documentation, including an example that was not running correctly

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by shadeMe** [comment]
Could you also update the outdated references to `Ragged` in `spaCy/website/docs/api/example.md`?

**Comment by richardpaulhudson** [comment]
Converted to draft as it turns out there is another aspect to this task I hadn't been aware of.

File: website/docs/api/example.md [hunk 1]
```diff
@@ -286,10 +286,12 @@ Calculate alignment tables between two tokenizations.
 
 ### Alignment attributes {#alignment-attributes"}
 
+Alignment attributes are managed using the `AlignmentArray`, which is a simplified version of Thinc's [Ragged](https://thinc.ai/docs/api-types#ragged) type that only supports the `data` and `length` attributes.
```
Comment (richardpaulhudson) [inline website/docs/api/example.md:289]: Adding a separate container entry for `AlignmentArray` seemed like overkill, so I just added this note instead. Although it closely follows the in-code documentation, it's not 100% accurate as `AlignmentArray` actually contains two additional properties, but none that a user would be likely to want to access. I feel this solution is a good compromise between accuracy and clarity/conciseness.

**Review by shadeMe — COMMENTED** [review]
Looks good! The example still contains a few references to `dataXd`, though (lines 314-317).

**Comment by richardpaulhudson** [comment]
Thanks, I hadn't seen those!

**Review by shadeMe — COMMENTED** [review]
LGTM! Just one minor comment.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**